src/NUnitFramework/tests/Constraints/RegexConstraintTests.cs

```diff
@@ -10,6 +10,17 @@ public class RegexConstraintTests
     {
         private static readonly string NL = Environment.NewLine;
 
+        [Test]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Assertion", "NUnit2024", Justification = "Testing this negative case")]
```

Comment:
There seem to be no other `SuppressMessage` calls in the code base.
Please replace it with:
```csharp
#pragma warning disable NUnit2024 // Wrong actual type used with String Constraint
...
#pragma warning restore NUnit2024 // Wrong actual type used with String Constraint
```